### DISCUSSION TITLE
Rustdesk build crash with error: failed to run custom build command

### DISCUSSION THREAD

**Discussion by Andre0be** [rustdesk/rustdesk#5313] [General]
Hello,

I want to build my own rustdesk with integrated server and key.

But I can't build an exe file. Unter Lubuntu and under fedora, after the command "VCPKG_ROOT=$HOME/vcpkg cargo run" I get this error:

`error: failed to run custom build command for `magnum-opus v0.4.0 (https://github.com/rustdesk/magnum-opus#5cd2bf98)`

Caused by:
  process didn't exit successfully: `/rustdesk/target/debug/build/magnum-opus-f1fdf099b5396614/build-script-build` (exit status: 101)
  --- stdout
  cargo:info=x64-linux
  cargo:rustc-link-lib=static=opus
  cargo:rustc-link-search=/root/vcpkg/installed/x64-linux/lib
  cargo:include=/root/vcpkg/installed/x64-linux/include
  rerun-if-changed=/root/.cargo/git/checkouts/magnum-opus-4bb999f3bcbf6ab0/5cd2bf9/opus_ffi.h
  rerun-if-changed=/root/vcpkg/installed/x64-linux/include

  --- stderr
  /root/.cargo/git/checkouts/magnum-opus-4bb999f3bcbf6ab0/5cd2bf9/opus_ffi.h:1:10: fatal error: 'opus/opus_multistream.h' file not found
  /root/.cargo/git/checkouts/magnum-opus-4bb999f3bcbf6ab0/5cd2bf9/opus_ffi.h:1:10: fatal error: 'opus/opus_multistream.h' file not found, err: true
  thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ()', /root/.cargo/git/checkouts/magnum-opus-4bb999f3bcbf6ab0/5cd2bf9/build.rs:146:18
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
`
I have search for this error, but I can't repair it. Where is the problem (except me)?

Thanks

**Comment by SergeyMy** [comment]
are you running as root? nooo so so solution
`vcpkg/vcpkg install libvpx libyuv opus aom`
did it?

**Comment by Andre0be** [comment]
yes, I build it with root.

I have followed the instructions:

https://rustdesk.com/docs/en/dev/build/linux/

I have made:
`vcpkg/vcpkg install libvpx libyuv opus aom`

**Comment by Andre0be** [comment]
With Ubuntu Server it works. But I can't create a exe or? What is the command to create a exe under ubuntu?

**Reply by deep-soft** [reply]
I see that this pinned thread it's not very visible:
[https://github.com/rustdesk/rustdesk/discussions/763](https://github.com/rustdesk/rustdesk/discussions/763)

Btw,
for .exe you must dig here:

> https://github.com/rustdesk/rustdesk/blob/master/.github/workflows/flutter-build.yml

first job is:
`build-for-windows-flutter:`

**Reply by Andre0be** [reply]
Can you explain it in more detail?

I can't get it to work.

Is there a tutorial?

I cant see the job "build-for-windows-flutter" in github. Is the real name "Build the frlutter version of the RustDesk"?
There is no worklow runs - I can't start it

**Reply by deep-soft** [reply]
If you open the link:
https://github.com/rustdesk/rustdesk/blob/master/.github/workflows/flutter-build.yml
from line 36 to line 138
you will find the job:
build-for-windows-flutter:

Of course before that you must learn what Github Actions are:
https://docs.github.com/en/actions

**Comment by Andre0be** [comment]
Unfortunately, I gave up. No instructions. RustDesk has potential, but building an exe is impossible (for beginners).

**Reply by beelchester** [reply]
Actually building an exe on a windows machine itself is quite easy.

If I remember correctly, it involved:
- Installing vcpkg (https://rustdesk.com/docs/en/dev/build/windows/#vcpkg)
- Generating flutter rust bridge files ([rustdesk/flutter/run.sh](https://github.com/rustdesk/rustdesk/blob/master/flutter/run.sh))
- Running `python3 build.py --flutter`

Might also need to get sciter.dll so for that:
```
mkdir -p target/debug
wget https://raw.githubusercontent.com/c-smile/sciter-sdk/master/bin.win/x64/sciter.dll
mv sciter.dll target/debug
```
(Source: https://rustdesk.com/docs/en/dev/build/windows/#default)

But I agree with you, it can be confusing for the beginners.

**Reply by SergeyMy** [reply]
everything is a little different

> Generating flutter rust bridge files (rustdesk/flutter/run.sh)

creates a flutter exe, and runs it :)
everything else is unnecessary (sciter branch)

**Reply by Andre0be** [reply]
...and how I can creates a flutter exe?

somehow this does not help me

is it easier with linux?

**Comment by SergeyMy** [comment]
as stated earlier,

> Of course before that you must learn what Github Actions are:
> https://docs.github.com/en/actions
> 

If you can’t build natively, trust github actions to do it! :) https://rustdesk.com/docs/en/dev/build/all/
1. fork rustedsk
2. Enable workflows 
3. Enable upload permissions for workflows 
4. run workflow  **Flutter Nightly Build** 
victory

**Comment by Andre0be** [comment]
Now I have build an exe file. But my server and my public key are not in the exe. RENDEZVOUS_SERVER and RS_PUB_KEY are set under Actions secrets and variables. But there is written: Anyone with collaborator access to this repository can use these secrets and variables for actions. They are not passed to workflows that are triggered by a pull request from a fork. 

how do i get the settings in the workflow?

**Reply by DarkZoneSD** [reply]
You fork the repository and edit your own Repository Secrets! You can't edit them in the official branch.
